integration/query_test.go

```diff
@@ -1053,16 +1053,48 @@ func TestQueryIDDoc(t *testing.T) {
 	})
 	require.NoError(t, err)
 
-	expected := []bson.D{{
-		{"_id", bson.D{{"a", int32(3)}, {"z", int32(4)}}},
-		{"v", int32(2)},
-	}}
-	actual := FilterAll(t, ctx, collection, bson.D{{"_id", bson.D{{"a", int32(3)}, {"z", int32(4)}}}})
-	AssertEqualDocumentsSlice(t, expected, actual)
+	t.Run("CorrectFieldOrder", func(t *testing.T) {
+		expected := []bson.D{{
+			{"_id", bson.D{{"a", int32(3)}, {"z", int32(4)}}},
+			{"v", int32(2)},
+		}}
+		actual := FilterAll(t, ctx, collection, bson.D{{"_id", bson.D{{"a", int32(3)}, {"z", int32(4)}}}})
+		AssertEqualDocumentsSlice(t, expected, actual)
+	})
+
+	t.Run("WrongFieldOrder", func(t *testing.T) {
+		expected := []bson.D{}
+		actual := FilterAll(t, ctx, collection, bson.D{{"_id", bson.D{{"z", int32(4)}, {"a", int32(3)}}}})
+		AssertEqualDocumentsSlice(t, expected, actual)
+	})
+
+	t.Run("PartialID", func(t *testing.T) {
+		cursor, err := collection.Find(ctx, bson.D{{"_id.a", int32(3)}})
+		require.NoError(t, err)
 
-	expected = []bson.D{}
-	actual = FilterAll(t, ctx, collection, bson.D{{"_id", bson.D{{"z", int32(4)}, {"a", int32(3)}}}})
-	AssertEqualDocumentsSlice(t, expected, actual)
+		expected := []bson.D{{
+			{"_id", bson.D{{"a", int32(3)}, {"z", int32(4)}}},
+			{"v", int32(2)},
+		}}
+		actual := FetchAll(t, ctx, cursor)
+		assert.Equal(t, expected, actual)
+	})
+
+	t.Run("Update", func(t *testing.T) {
+		res, err := collection.UpdateOne(ctx, bson.D{{"_id.a", int32(1)}}, bson.D{{"$set", bson.D{{"v", int32(3)}}}})
```

Comment:
It looks like we completely ignore our own test naming guidelines and update tests to query_test.go